### PR TITLE
Remove non-python setup from a number of machine setup scripts

### PR DESCRIPTION
**Pull request by jgfouca**
CIME only requires python to run. Loading anything else in these
files is misleading because the first thing CIME does on many machines
is a module purge, so loading cmake, compilers, or other things here has no impact
on what CIME actually uses on the code.

### CODE REVIEW

**Review by amametjanov — APPROVED** [review]

File: util/anvil_setup.sh [hunk 1]
```diff
@@ -4,6 +4,4 @@ source  /lcrc/soft/climate/e3sm-unified/load_latest_cime_env.sh
 source /home/software/spack-0.10.1/opt/spack/linux-centos7-x86_64/gcc-4.8.5/lmod-7.4.9-ic63herzfgw5u3na5mdtvp3nwxy6oj2z/lmod/lmod/init/sh
 export MODULEPATH=$MODULEPATH:/software/centos7/spack-latest/share/spack/lmod/linux-centos7-x86_64/Core
 
-module load intel/17.0.0-pwabdn2 cmake/3.14.2-gvwazz3 git/2.13.0
```
Comment (rljacob) [inline util/anvil_setup.sh:7]: the git module might be needed.

Comment (rljacob) [reply util/anvil_setup.sh:7]: Default git on blues is  1.8.3.1

Comment (jgfouca) [reply util/anvil_setup.sh:7]: @rljacob , what is the context where you think this is needed? Jenkins must have Git to do any repo stuff at the beginning of the job (can't source stuff from E3SM_test_scripts if you can't clone it), so usually you tell the various nodes where to find git in their configurations.

Comment (rljacob) [reply util/anvil_setup.sh:7]: That git version is enough to clone the scripts but maybe not enough to clone E3SM with recursive submodules?  I'm not sure where the clone of E3SM is in relation to this line.

Comment (bartgol) [reply util/anvil_setup.sh:7]: FWIW, I had issues with the alias in my gitconfig for submodule updates on some systems:

```
git submodule update --init --recursive --progress
```
In particular, older version of git don't recognize the `--progress` option. Maybe something like this is happening...

Comment (jgfouca) [reply util/anvil_setup.sh:7]: @rljacob , @bartgol , ALL jenkins Git actions occur before this line is seen.

**Pull request merged**